src/main/java/org/folio/rest/service/action/ActionService.java

```diff
@@ -168,4 +179,16 @@ protected Feefineaction createFeeFineActionAndUpdateAccount(Account account, Mon
     return succeededFuture(context);
   }
 
+  private Future<ActionContext> publishLogEvents(ActionContext actionContext) {
+    return all(actionContext.getFeeFineActions().stream()
+      // do not publish log records for CREDIT and REFUND actions
+      .filter(ffa -> !CREDIT.isActionForResult(ffa.getTypeAction()) && !REFUND.isActionForResult(ffa.getTypeAction()))
+      .map(ffa -> logEventService.createFeeFineLogEventPayload(ffa, actionContext.getAccounts().get(ffa.getAccountId()))
+        .compose(eventPayload -> {
+          logEventPublisher.publishLogEvent(eventPayload, FEE_FINE);
```

Comment:
Is this published async? Maybe it makes sense to make CompletableFuture.runAsync() and verify somehow by introduction of Thread.sleep()?